### PR TITLE
tiflash: add doc for tiflash pipeline model

### PR DESCRIPTION
**Pull request by SeaRise**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-pipeline-model.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+- 如果需要在当前 session 中开启 TiFlash Pipeline Model，可以通过以下语句设置：
+
+    ```sql
+    SET SESSION tidb_enable_tiflash_pipeline_model=ON;
+    ```
+
+- 如果需要在 global 级别开启 TiFlash Pipeline Model，可以通过以下语句设置：
+
+    ```sql
+    SET GLOBAL tidb_enable_tiflash_pipeline_model=OFF;
```
Comment (SeaRise) [inline tiflash/tiflash-pipeline-model.md:53]:
```suggestion
    SET GLOBAL tidb_enable_tiflash_pipeline_model=ON;
```

File: system-variables.md [hunk 2]
```diff
@@ -2528,6 +2528,21 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量用来控制是否启用 TiFlash 新的执行模型 [Pipeline Model](/tiflash/tiflash-pipeline-model.md)。
+- 当设置该变量为 `OFF` 关闭 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 原有的执行模型 Stream Model 来执行。当设置该变量为 `ON` 开启 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 新的执行模型 Pipeline Model 来执行。
+
+> **注意：**
+>
+> - TiFlash Pipeline Model 目前为实验特性，不建议在生产环境中使用。
+> - TiFlash Pipeline Model 目前不支持 [join 算子落盘](/tiflash/tiflash-spill-disk.md)，在开启 join 算子落盘且 `tidb_enable_tiflash_pipeline_model` 设置为 `ON` 时，下推到 tiflash 的查询仍会使用原有的执行模型 Stream Model 来执行。
+> - TiFlash Pipeline Model 目前不支持 [TiFlash 存算分离架构与 S3](/tiflash/tiflash-disaggregated-and-s3.md)，在开启 TiFlash 存算分离架构与 S3 且 `tidb_enable_tiflash_pipeline_model` 设置为 `ON` 时，下推到 tiflash 的查询仍会使用原有的执行模型 Stream Model 来执行。
```
[the remark this answers is not in the document]
Comment (SeaRise) [reply system-variables.md]: > 2. 是否要在 [数据落盘](https://github.com/pingcap/docs-cn/blob/5e27177c9f3c2c57727253e4c224f0a435981a5e/tiflash/tiflash-spill-disk.md#%E6%B3%A8%E6%84%8F)的**注意**和[存算分离](https://github.com/pingcap/docs-cn/blob/5e27177c9f3c2c57727253e4c224f0a435981a5e/tiflash/tiflash-disaggregated-and-s3.md#%E4%BD%BF%E7%94%A8%E9%99%90%E5%88%B6)的**使用限制**里，说明不支持 pipeline model

感觉不需要，因为无论开不开 pipeline model，这两个功能都会正常运行，只是说 pipeline model 会不会实际生效，但是对用户来说没影响。

[the remark this answers is not in the document]
Comment (SeaRise) [reply system-variables.md]: 那 join 落盘不支持那个我换一个链接
(/system-variables.md#tidb_max_bytes_before_tiflash_external_join-从-v700-版本开始引入)
这个直接指向了 join 落盘那个参数

File: tiflash/tiflash-pipeline-model.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+---
+title: TiFlash Pipeline Model 执行模型
+summary: 介绍 TiFlash 新的执行模型 Pipeline Model。
+---
+
+# TiFlash Pipeline Model
+
```
[the remark this answers is not in the document]
Comment (SeaRise) [reply tiflash/tiflash-pipeline-model.md:13]: 好的，加上了

File: tiflash/tiflash-pipeline-model.md
[the remark this answers is not in the document]
Comment (SeaRise) [reply tiflash/tiflash-pipeline-model.md:1]: 好的，已添加

File: system-variables.md [hunk 4]
```diff
@@ -2528,6 +2528,21 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 > - 假设一个 TiDB 集群有多个 TiFlash 节点，TopN 和 Sort 通常会在多个 TiFlash 节点中分布式执行。该变量控制的是单个 TiFlash 节点中 TopN 和 Sort 算子的最大内存使用量。
 > - 当该变量设置为 -1 时，TiFlash 将根据自身配置项 [`max_bytes_before_external_sort`](/tiflash/tiflash-configuration.md#tiflash-配置参数-1) 的值来决定 TopN 和 Sort 算子的最大内存使用量。
 
+### `tidb_enable_tiflash_pipeline_model` <span class="version-mark">从 v7.2.0 版本开始引入</span>
```
[the remark this answers is not in the document]
Comment (SeaRise) [reply system-variables.md]: 好的，done

File: tiflash/tiflash-pipeline-model.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+- 查询会被划分为多个 pipeline 并依次执行。在每个 pipeline 中，数据块会被尽可能保留在缓存中，从而实现更好的时间局部性，从而提高整个执行过程的效率。
+
+- 为了摆脱操作系统原生的线程调度模型，实现更加精细的调度机制，每个 pipeline 会被实例化成若干个 task，使用 task 调度模型，同时使用固定线程池，减少了操作系统申请和调度线程的开销。
+
+TiFlash Pipeline Model 的架构如下：
+
+![TiFlash Pipeline Model Design](/media/tiflash/tiflash-pipeline-model.png)
+
+如上图所示，Pipline Model 中有两个主要组成部分：Pipeline Query Executor 和 Task Scheduler。
```
Comment (ran-huang) [inline tiflash/tiflash-pipeline-model.md:96]:
```suggestion
如上图所示，Pipeline Model 中有两个主要组成部分：Pipeline Query Executor 和 Task Scheduler。
```

File: tiflash/tiflash-pipeline-model.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+    在查询被转换为 pipeline dag 后，Pipeline Query Executor 会按照依赖关系依次执行每个 pipeline。pipeline 会根据查询并发度被实例化成若干个 task 提交给 Task Scheduler 执行。
+
+- Task Scheduler
+
+    负责执行由 Pipeline Query Executor 提交过来的 task。task 会根据执行的逻辑的不同，在 Task Scheduler 里的不同组件中动态切换执行。
+
+    - CPU Task Thread Pool
+
+      执行 task 中 cpu 密集型的计算逻辑，比如数据过滤，函数计算等等。
```
Comment (ran-huang) [inline tiflash/tiflash-pipeline-model.md:114]:
```suggestion
      执行 task 中 CPU 密集型的计算逻辑，比如数据过滤、函数计算等。
```

File: tiflash/tiflash-pipeline-model.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+    - IO Task Thread Pool
+
+      执行 task 中 io 密集型的计算逻辑，比如计算中间结果落盘等等。
```
Comment (ran-huang) [inline tiflash/tiflash-pipeline-model.md:118]:
```suggestion
      执行 task 中 IO 密集型的计算逻辑，比如计算中间结果落盘等。
```

File: tiflash/tiflash-pipeline-model.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+    - Wait Reactor
+
+      执行 task 中的等待逻辑，比如等待网络层将数据包传输给计算层等等。
```
Comment (ran-huang) [inline tiflash/tiflash-pipeline-model.md:122]:
```suggestion
      执行 task 中的等待逻辑，比如等待网络层将数据包传输给计算层等。
```

**Review by ran-huang — COMMENTED** [review]
rest LGTM

File: system-variables.md [hunk 9]
```diff
@@ -1932,6 +1932,23 @@ Query OK, 0 rows affected (0.09 sec)
...
 - 这个变量用于动态地控制 TiDB 遥测功能是否开启，当前版本默认关闭 TiDB 的遥测功能。当所有 TiDB 实例都设置配置项 [`enable-telemetry`](/tidb-configuration-file.md#enable-telemetry-从-v402-版本开始引入) 为 `false` 时，将忽略该系统变量，并总是关闭 TiDB 遥测功能。参阅[遥测](/telemetry.md)了解该功能详情。
 
+### `tidb_enable_tiflash_pipeline_model` <span class="version-mark">从 v7.2.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量用来控制是否启用 TiFlash 新的执行模型 [Pipeline Model](/tiflash/tiflash-pipeline-model.md)。
+- 当设置该变量为 `OFF` 关闭 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 原有的执行模型 Stream Model 来执行。当设置该变量为 `ON` 开启 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 新的执行模型 Pipeline Model 来执行。
```
Comment (hfxsd) [inline system-variables.md:1942]:
```suggestion
- 当设置该变量为 `OFF` 关闭 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 原有的执行模型 Stream Model 来执行。
- 当设置该变量为 `ON` 开启 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 新的执行模型 Pipeline Model 来执行。
```

**Review by hfxsd — APPROVED** [review]
Rest LGTM

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 10]
```diff
@@ -1932,6 +1932,24 @@ Query OK, 0 rows affected (0.09 sec)
...
+- 这个变量用来控制是否启用 TiFlash 新的执行模型 [Pipeline Model](/tiflash/tiflash-pipeline-model.md)。
+- 当设置该变量为 `OFF` 关闭 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 原有的执行模型 Stream Model 来执行。
+- 当设置该变量为 `ON` 开启 TiFlash Pipeline Model 时，下推到 TiFlash 的查询会使用 TiFlash 新的执行模型 Pipeline Model 来执行。
+
+> **注意：**
+>
+> - TiFlash Pipeline Model 目前为实验特性，不建议在生产环境中使用。
+> - TiFlash Pipeline Model 目前不支持以下功能。当下列功能开启时，即使 `tidb_enable_tiflash_pipeline_model` 设置为 `ON`，下推到 TiFlash 的查询仍会使用原有的执行模型 Stream Model 来执行。
+>
+>     - [Join 算子落盘](/system-variables.md#tidb_max_bytes_before_tiflash_external_join-从-v700-版本开始引入)
```
Comment (ran-huang) [inline system-variables.md:1950]:
```suggestion
>     - [Join 算子落盘](#tidb_max_bytes_before_tiflash_external_join-从-v700-版本开始引入)
```

**Review by joey-yez — APPROVED** [review]

**Pull request merged**